daemon/api.go

```diff
@@ -778,6 +781,7 @@ func snapInstall(inst *snapInstruction, st *state.State) (string, []*state.TaskS
 }
 
 func snapUpdate(inst *snapInstruction, st *state.State) (string, []*state.TaskSet, error) {
+	// TODO: bail if revision is given (and != current?), *or* behave as with install --revision?
```

Comment:
> The basic semantics of install and refresh should be preserved though: install does not work if the snap is already installed. Refresh doesn't work if the snap is not installed.

Well... I kinda disagree there a little bit. Because sideloads: `snap install ./foo.snap` and `snap install --revision nn foo` to me should behave about the same way, except the second one goes through the store.

That *might* mean that we block the first one with an "already installed" message and implement `snap refresh ./foo`, but that feels even weirder.

Anyhow, yes, `snap refresh --revision nn foo` is coming in a follow-up.